src/main.ts

```diff
@@ -1212,21 +1212,17 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
       const fallback = isWithinOneOf(field) ? 'undefined' : defaultValue(ctx, field);
       chunks.push(code`message.${fieldName} = ${fallback}`);
       chunks.push(code`}`);
-    } else if (
-      isPrimitive(field) ||
-      (isTimestamp(field) && (options.useDate === DateOption.DATE || options.useDate === DateOption.STRING)) ||
-      isValueType(ctx, field)
-    ) {
+    } else if (readSnippet(`x`).toCodeString() == 'x') {
```

Comment:
This somehwat unrelated change is worth having a 👀.

Reply:
Ah yeah, that's a neat way of doing this.